File: test/leetcode/m97/InterleavingStringTest.java
/*
 * Copyright (c) 2024 THGEEK.COM. All rights reserved.
 */

package leetcode.m97;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

/**
 * Test for {@link InterleavingString}
 *
 * @author Tao Hong
 * @version 1.0
 * @since 2024/10/28 12:54
 */
class InterleavingStringTest {
    private InterleavingString test;

    @BeforeEach
    void setUp() {
        test = new InterleavingString();
    }

    @Test
    void isTrue() {
        String s1 = "aabcc";
        String s2 = "dbbca";
        String s3 = "aadbbcbcac";
        assertTrue(test.isInterleave(s1, s2, s3));
    }

    @Test
    void isFalse() {
        String s1 = "aabcc";
        String s2 = "dbbca";
        String s3 = "aadbbbaccc";
        assertFalse(test.isInterleave(s1, s2, s3));
    }

    @Test
    void isTrue2() {
        String s1 = "";
        String s2 = "";
        String s3 = "";
        assertTrue(test.isInterleave(s1, s2, s3));
    }

    @Test
    void isFalse2() {
        String s1 = "a";
        String s2 = "";
        String s3 = "c";
        assertFalse(test.isInterleave(s1, s2, s3));
    }

    @Test
    void isTrue3() {
        String s1 = "aa";
        String s2 = "ab";
        String s3 = "abaa";
        assertTrue(test.isInterleave(s1, s2, s3));
    }
}

File: src/leetcode/m97/InterleavingString_test.go
package m97

import "testing"

func Test_isInterleave(t *testing.T) {
	type args struct {
		s1 string
		s2 string
		s3 string
	}
	tests := []struct {
		name string
		args args
		want bool
	}{
		{"Test case 1", args{"aabcc", "dbbca", "aadbbcbcac"}, true},
		{"Test case 2", args{"aabcc", "dbbca", "aadbbbaccc"}, false},
		{"Test case 3", args{"", "", ""}, true},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := isInterleave(tt.args.s1, tt.args.s2, tt.args.s3); got != tt.want {
				t.Errorf("isInterleave() = %v, want %v", got, tt.want)
			}
		})
	}
}
